### DISCUSSION TITLE
About 3D model air supply and exhaust in FDS

### DISCUSSION THREAD

**Discussion by joker121-move** [firemodels/fds#13939] [General]
Hi experts, I am simulating about DS-TJ air curtain, i.e. a device where one end is sucked in through mechanical exhaust and then discharged again, and one end is sucked in through the outside world and then discharged again. However, during the simulation, I found that my setup is not getting the result I want (not being able to exhaust as well as suck in air at the air outlet)， may be it is due to my ducting setup? I would like to ask you guys how to set up this 3D model in a reasonable way? Thank you very much!
The grid is set small as it is still in the testing phase.

The grid is set small as it is still in the testing phase

&HEAD CHID='extinguish'/
&TIME T_END=500.0/
&DUMP RENDER_FILE='extinguish.ge1', DT_RESTART=1.0, DT_SL3D=0.1/
&MISC CHECK_HT=.TRUE., TURBULENCE_MODEL='DYNAMIC SMAGORINSKY'/
&CLIP CLIP_DT_RESTRICTIONS_MAX=0/

&MESH ID='Mesh01', IJK=5,8,7, XB=0.0,5.0,0.0,8.0,0.0,7.0/
&MESH ID='Mesh02', IJK=5,8,7, XB=5.0,10.0,0.0,8.0,0.0,7.0/
&MESH ID='Mesh03', IJK=5,8,7, XB=10.0,15.0,0.0,8.0,0.0,7.0/
&MESH ID='Mesh04', IJK=5,8,7, XB=15.0,20.0,0.0,8.0,0.0,7.0/
&MESH ID='Mesh05', IJK=5,8,7, XB=20.0,25.0,0.0,8.0,0.0,7.0/
&MESH ID='Mesh06', IJK=5,8,7, XB=25.0,30.0,0.0,8.0,0.0,7.0/
&MESH ID='Mesh07', IJK=5,8,7, XB=30.0,35.0,0.0,8.0,0.0,7.0/
&MESH ID='Mesh08', IJK=5,8,7, XB=35.0,40.0,0.0,8.0,0.0,7.0/
&MESH ID='Mesh09', IJK=5,8,7, XB=40.0,45.0,0.0,8.0,0.0,7.0/
&MESH ID='Mesh10', IJK=5,8,7, XB=45.0,50.0,0.0,8.0,0.0,7.0/
&MESH ID='Mesh11', IJK=5,8,7, XB=50.0,55.0,0.0,8.0,0.0,7.0/
&MESH ID='Mesh12', IJK=5,8,7, XB=55.0,60.0,0.0,8.0,0.0,7.0/
&MESH ID='Mesh13', IJK=5,8,7, XB=60.0,65.0,0.0,8.0,0.0,7.0/
&MESH ID='Mesh14', IJK=5,8,7, XB=65.0,70.0,0.0,8.0,0.0,7.0/
&MESH ID='Mesh15', IJK=5,8,7, XB=70.0,75.0,0.0,8.0,0.0,7.0/
&MESH ID='Mesh16', IJK=5,8,7, XB=75.0,80.0,0.0,8.0,0.0,7.0/
&MESH ID='Mesh17', IJK=5,8,10, XB=80.0,85.0,0.0,8.0,0.0,10.0/
&MESH ID='Mesh18', IJK=5,8,10, XB=85.0,90.0,0.0,8.0,0.0,10.0/

&REAC ID='HEPTANE',
      FYI='NIST NRC FDS5 Validation',
      FUEL='REAC_FUEL',
      FORMULA='C7H16',
      CO_YIELD=6.0E-3,
      SOOT_YIELD=0.015,
      RADIATIVE_FRACTION=0.35/

&DEVC ID='GAS', QUANTITY='VELOCITY', XYZ=49.6,1.0,4.5/
&DEVC ID='GAS01', QUANTITY='VELOCITY', XYZ=49.6,2.0,4.5/
&DEVC ID='GAS02', QUANTITY='VELOCITY', XYZ=49.6,3.0,4.5/
&DEVC ID='GAS03', QUANTITY='VELOCITY', XYZ=49.6,4.0,4.5/
&DEVC ID='GAS04', QUANTITY='VELOCITY', XYZ=49.6,5.0,4.5/
&DEVC ID='GAS05', QUANTITY='VELOCITY', XYZ=49.6,6.0,4.5/
&DEVC ID='GAS06', QUANTITY='VELOCITY', XYZ=49.6,7.0,4.5/
&DEVC ID='GAS07', QUANTITY='VELOCITY', XYZ=49.6,8.0,4.5/
&DEVC ID='GAS08', QUANTITY='VELOCITY', XYZ=50.4,1.0,4.5/
&DEVC ID='GAS09', QUANTITY='VELOCITY', XYZ=50.4,2.0,4.5/
&DEVC ID='GAS10', QUANTITY='VELOCITY', XYZ=50.4,3.0,4.5/
&DEVC ID='GAS11', QUANTITY='VELOCITY', XYZ=50.4,4.0,4.5/
&DEVC ID='GAS12', QUANTITY='VELOCITY', XYZ=50.4,5.0,4.5/
&DEVC ID='GAS13', QUANTITY='VELOCITY', XYZ=50.4,6.0,4.5/
&DEVC ID='GAS14', QUANTITY='VELOCITY', XYZ=50.4,7.0,4.5/
&DEVC ID='GAS15', QUANTITY='VELOCITY', XYZ=50.4,8.0,4.5/
&DEVC ID='GAS16', QUANTITY='VELOCITY', XYZ=19.5,4.0,4.5/

&MATL ID='CONCRETE',
      FYI='NBSIR 88-3752 - ATF NIST Multi-Floor Validation',
      SPECIFIC_HEAT=1.04,
      CONDUCTIVITY=1.8,
      DENSITY=2280.0/
&MATL ID='STEEL',
      FYI='Drysdale, Intro to Fire Dynamics - ATF NIST Multi-Floor Validation',
      SPECIFIC_HEAT=0.46,
      CONDUCTIVITY=45.8,
      DENSITY=7850.0,
      EMISSIVITY=0.95/

&SURF ID='concrete',
      COLOR='GRAY 60',
      BACKING='VOID',
      MATL_ID(1,1)='CONCRETE',
      MATL_MASS_FRACTION(1,1)=1.0,
      THICKNESS(1)=0.01/
&SURF ID='pipe',
      RGB=204,255,255,
      BACKING='VOID',
      MATL_ID(1,1)='STEEL',
      MATL_MASS_FRACTION(1,1)=1.0,
      THICKNESS(1)=0.01/
&SURF ID='fire',
      COLOR='RED',
      HRRPUA=2.0E4,
      TMP_FRONT=300.0/

&OBST ID='ceiling', XB=0.0,80.0,0.0,8.0,4.5,4.5, SURF_ID='concrete'/ 
&OBST ID='end', XB=0.0,0.0,0.0,8.0,0.0,4.5, SURF_ID='concrete'/ 
&OBST ID='right', XB=0.0,80.0,0.0,0.0,0.0,4.5, COLOR='INVISIBLE', SURF_ID='concrete'/ 
&OBST ID='left', XB=0.0,80.0,8.0,8.0,0.0,4.5, COLOR='INVISIBLE', SURF_ID='concrete'/ 
&OBST ID='flood', XB=0.0,80.0,0.0,8.0,0.0,0.0, SURF_ID='concrete'/ 
&OBST ID='Pipe', XB=19.0,19.0,0.0,8.0,4.5,7.0, SURF_ID='pipe'/ 
&OBST ID='Pipe', XB=20.0,20.0,0.0,8.0,4.5,6.0, SURF_ID='pipe'/ 
&OBST ID='Pipe', XB=50.0,50.0,0.0,8.0,4.5,7.0, SURF_ID='pipe'/ 
&OBST ID='Pipe', XB=49.2,49.2,0.0,8.0,4.5,6.0, SURF_ID='pipe'/ 
&OBST ID='Pipe', XB=49.2,50.0,0.0,0.0,4.5,7.0, COLOR='INVISIBLE', SURF_ID='pipe'/ 
&OBST ID='Pipe', XB=49.2,50.0,8.0,8.0,4.5,7.0, SURF_ID='pipe'/ 
&OBST ID='Pipe', XB=19.0,49.2,3.5,3.5,6.0,7.0, COLOR='INVISIBLE', SURF_ID='pipe', TEXTURE_ORIGIN=19.0,3.5,6.0/ 
&OBST ID='Pipe', XB=20.0,49.2,0.0,8.0,6.0,6.0, SURF_ID='pipe'/ 
&OBST ID='Pipe', XB=19.0,49.2,0.0,8.0,7.0,7.0, SURF_ID='pipe'/ 
&OBST ID='Pipe', XB=49.2,50.0,0.0,8.0,7.0,7.0, SURF_ID='pipe'/ 
&OBST ID='Pipe', XB=50.0,50.8,0.0,0.0,4.5,7.0, COLOR='INVISIBLE', SURF_ID='pipe'/ 
&OBST ID='Pipe', XB=50.0,50.8,0.0,8.0,7.0,7.0, SURF_ID='pipe'/ 
&OBST ID='Pipe', XB=50.0,50.0,0.0,8.0,4.5,6.0, SURF_ID='pipe'/ 
&OBST ID='Pipe', XB=50.0,50.0,0.0,3.5,6.0,7.0, SURF_ID='pipe'/ 
&OBST ID='Pipe', XB=50.0,50.0,4.5,8.0,6.0,7.0, SURF_ID='pipe'/ 
&OBST ID='Pipe', XB=50.8,50.8,0.0,8.0,4.5,6.0, SURF_ID='pipe'/ 
&OBST ID='Pipe', XB=50.8,85.0,0.0,8.0,7.0,7.0, SURF_ID='pipe'/ 
&OBST ID='Pipe', XB=50.8,85.0,3.5,3.5,6.0,7.0, COLOR='INVISIBLE', SURF_ID='pipe', TEXTURE_ORIGIN=50.8,3.5,6.0/ 
&OBST ID='Pipe', XB=50.8,85.0,0.0,8.0,6.0,6.0, SURF_ID='pipe'/ 
&OBST ID='Pipe', XB=80.0,80.0,0.0,8.0,4.5,6.0, SURF_ID='pipe'/ 
&OBST ID='Pipe', XB=50.0,50.8,8.0,8.0,4.5,7.0, SURF_ID='pipe'/ 
&OBST ID='fengji', XB=29.0,30.0,0.0,8.0,6.0,7.0, SURF_ID='INERT'/ 
&OBST ID='fengji', XB=84.0,85.0,0.0,8.0,6.0,7.0, SURF_ID='INERT'/ 

&HOLE ID='exhaust', XB=19.0,20.1,-0.1,8.1,4.5,4.5/ 
&HOLE ID='supply1', XB=49.2,50.1,-0.1,8.1,4.5,4.5/ 
&HOLE ID='supply1', XB=49.9,50.8,-0.1,8.1,4.5,4.5/ 

&VENT ID='Mesh Vent: Mesh17 [XMIN]', SURF_ID='OPEN', XB=80.0,80.0,0.0,8.0,7.0,10.0/ 
&VENT ID='Mesh Vent: Mesh17 [YMAX]', SURF_ID='OPEN', XB=80.0,85.0,8.0,8.0,0.0,10.0/ 
&VENT ID='Mesh Vent: Mesh17 [YMIN]', SURF_ID='OPEN', XB=80.0,85.0,0.0,0.0,0.0,10.0/ 
&VENT ID='Mesh Vent: Mesh17 [ZMAX]', SURF_ID='OPEN', XB=80.0,85.0,0.0,8.0,10.0,10.0/ 
&VENT ID='Mesh Vent: Mesh17 [ZMIN]', SURF_ID='OPEN', XB=80.0,85.0,0.0,8.0,0.0,0.0/ 
&VENT ID='Mesh Vent: Mesh18 [XMAX]', SURF_ID='OPEN', XB=90.0,90.0,0.0,8.0,0.0,10.0/ 
&VENT ID='Mesh Vent: Mesh18 [YMAX]', SURF_ID='OPEN', XB=85.0,90.0,8.0,8.0,0.0,10.0/ 
&VENT ID='Mesh Vent: Mesh18 [YMIN]', SURF_ID='OPEN', XB=85.0,90.0,0.0,0.0,0.0,10.0/ 
&VENT ID='Mesh Vent: Mesh18 [ZMAX]', SURF_ID='OPEN', XB=85.0,90.0,0.0,8.0,10.0,10.0/ 
&VENT ID='Mesh Vent: Mesh18 [ZMIN]', SURF_ID='OPEN', XB=85.0,90.0,0.0,8.0,0.0,0.0/ 
&VENT ID='Vent05', SURF_ID='HVAC', XB=84.0,84.0,3.5,4.5,6.0,7.0/ 
&VENT ID='Vent06', SURF_ID='HVAC', XB=85.0,85.0,0.0,8.0,6.0,7.0/ 
&VENT ID='Vent02', SURF_ID='HVAC', XB=29.0,29.0,0.0,8.0,6.0,7.0/ 
&VENT ID='Vent03', SURF_ID='HVAC', XB=30.0,30.0,0.0,8.0,6.0,7.0/ 
&VENT ID='Vent01', SURF_ID='fire', XB=4.5,5.5,3.5,4.5,0.0,0.0/ 

&HVAC ID='Node03', TYPE_ID='NODE', DUCT_ID='Duct02', VENT_ID='Vent05'/
&HVAC ID='Node04', TYPE_ID='NODE', DUCT_ID='Duct02', VENT_ID='Vent06'/
&HVAC ID='Duct02', TYPE_ID='DUCT', DIAMETER=0.3048, VOLUME_FLOW=100.0, NODE_ID='Node04','Node03', ROUGHNESS=0.0/
&HVAC ID='Node01', TYPE_ID='NODE', DUCT_ID='Duct01', VENT_ID='Vent02'/
&HVAC ID='Node02', TYPE_ID='NODE', DUCT_ID='Duct01', VENT_ID='Vent03'/
&HVAC ID='Duct01', TYPE_ID='DUCT', DIAMETER=0.3048, VOLUME_FLOW=100.0, NODE_ID='Node01','Node02', ROUGHNESS=0.0/

&SLCF QUANTITY='TEMPERATURE', PBY=4.0/
&SLCF QUANTITY='VELOCITY', VECTOR=.TRUE., PBY=4.0/
&SLCF QUANTITY='VOLUME FRACTION', SPEC_ID='AIR', PBY=4.0/
&SLCF QUANTITY='VOLUME FRACTION', SPEC_ID='OXYGEN', PBY=4.0/
&SLCF QUANTITY='VOLUME FRACTION', SPEC_ID='REAC_FUEL', PBY=4.0/
&SLCF QUANTITY='PRESSURE', PBY=4.0/

&TAIL /
<img width="652" alt="help" src="https://github.com/user-attachments/assets/e2b84047-620d-4bea-961e-ee192052f577" />

**Comment by Er9y714** [comment]
![image](https://github.com/user-attachments/assets/fa509088-da8e-4cea-ab29-dd8dcf27aef7)

- The setup is made for older versions. You should adjust the input to run the simulation in the latest version (6.9.1)
- I would recommend simplifying the setup and go in steps. e.g. verify HVAC is working, then adding fire etc. 
- There are OBSTructions and VENTs which are not aligning with the mesh. They need to be fixed (see image above)
- The mesh resolution is too coarse. For developing more realistic velocity profiles, you should consider making it finer, once you fix the HVAC module.

**Reply by joker121-move** [reply]
Thank you very much for your help, I have fixed the model under your guidance and have now implemented the feature, thanks again!
